server/integration/endpoint/src/main/java/org/infinispan/server/endpoint/subsystem/RouterSubsystemAdd.java

```diff
@@ -0,0 +1,83 @@
+import org.jboss.as.controller.AbstractAddStepHandler;
+import org.jboss.as.controller.AttributeDefinition;
+import org.jboss.as.controller.OperationContext;
+import org.jboss.as.controller.OperationFailedException;
+import org.jboss.as.controller.PathAddress;
+import org.jboss.as.controller.registry.Resource;
+import org.jboss.dmr.ModelNode;
+import org.jboss.msc.service.ServiceBuilder;
+import org.jboss.msc.service.ServiceName;
+
+import java.util.Optional;
+
+class RouterSubsystemAdd extends AbstractAddStepHandler {
+
+   static final RouterSubsystemAdd INSTANCE = new RouterSubsystemAdd();
+
+   @Override
+   protected void populateModel(ModelNode source, ModelNode target) throws OperationFailedException {
+      for(AttributeDefinition attr : RouterConnectorResource.ROUTER_CONNECTOR_ATTRIBUTES) {
+         attr.validateAndSet(source, target);
+      }
+   }
+
+   @Override
+   protected void performRuntime(OperationContext context, ModelNode operation, ModelNode model) throws OperationFailedException {
+      // Read the full model
+      ModelNode config = Resource.Tools.readModel(context.readResource(PathAddress.EMPTY_ADDRESS));
+
+      MultiTenantRouterConfigurationBuilder configuration = new MultiTenantRouterConfigurationBuilder();
+      MultiTenantRouterService routerService = new MultiTenantRouterService(configuration, getServiceName(config));
+
+      final ServiceName MultitenantRouterServiceName = EndpointUtils.getServiceName(operation, "router");
+      ServiceBuilder<?> builder = context.getServiceTarget().addService(MultitenantRouterServiceName, routerService);
+      EndpointUtils.addSocketBindingDependency(builder, getSocketBindingName(operation), routerService.getSocketBinding());
+
+      if(config.get(ModelKeys.HOTROD).isDefined()) {
+         config.get(ModelKeys.HOTROD).asList().forEach(hr -> {
+            String hotRodName = hr.get(0).get(ModelKeys.NAME).asString();
+            if(hr.get(0).get(ModelKeys.SNI).isDefined()) {
+               hr.get(0).get(ModelKeys.SNI).asList().forEach(sni -> {
+                  String sniHostName = sni.get(0).get(ModelKeys.HOST_NAME).asString();
+                  String securityRealm = sni.get(0).get(ModelKeys.SECURITY_REALM).asString();
+                  MultiTenantRouterService.HotRodRouting hotRodRouting = routerService.getHotRodRouting(sniHostName);
+                  EndpointUtils.addHotRodDependency(builder, hotRodName, hotRodRouting.getHotRod());
+                  EndpointUtils.addSecurityRealmDependency(builder, securityRealm, hotRodRouting.getSecurityRealm());
+               });
+            }
+         });
+      }
+
+      builder.install();
+   }
+
+   private Optional<String> getServiceName(ModelNode config) {
+      return Optional.ofNullable(config.get(ModelKeys.NAME)).map(m -> m.asString());
+   }
+
+   private String getSocketBindingName(ModelNode config) {
+      return Optional.of(config.get(ModelKeys.SOCKET_BINDING)).map(m -> m.asString()).get();
```

Comment:
Seems this could just be `config.get(ModelKeys.SOCKET_BINDING).asString()` or did you mean `ofNullable`?